File: src/main/java/at/typischersepp/fortnite/handlers/TMP.java
package at.typischersepp.fortnite.handlers;

import at.typischersepp.fortnite.Fortnite;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.NamespacedKey;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.persistence.PersistentDataType;

public class TMP implements Listener {
    Fortnite plugin;

    public TMP(Fortnite plugin) {
        this.plugin = plugin;
        Bukkit.getPluginManager().registerEvents(this, plugin);
    }

    @EventHandler
    public void onJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();

        ItemStack gun = new ItemStack(Material.STICK);
        NamespacedKey key = new NamespacedKey(plugin, "ammo");
        ItemMeta itemMeta = gun.getItemMeta();
        itemMeta.getPersistentDataContainer().set(key, PersistentDataType.INTEGER, 10);

        gun.setItemMeta(itemMeta);

        player.getInventory().setItemInMainHand(gun);
    }
}

File: src/main/java/at/typischersepp/fortnite/handlers/TMP.kt
package at.typischersepp.fortnite.handlers

import at.typischersepp.fortnite.Fortnite
import at.typischersepp.fortnite.utils.setMetadata
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.NamespacedKey
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.inventory.ItemStack
import org.bukkit.persistence.PersistentDataType

class TMP(private var plugin: Fortnite) : Listener {
    // Initialize Handler
    init {
        Bukkit.getPluginManager().registerEvents(this, plugin)
    }

    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        // Initialize variables for better understanding
        val player = event.player
        val gun = ItemStack(Material.STICK)

        // Set metadata to gun
        setMetadata(gun, NamespacedKey(plugin, "ammo"), PersistentDataType.INTEGER, 20)
        setMetadata(gun, NamespacedKey(plugin, "ammoCapacity"), PersistentDataType.INTEGER, 20)
        setMetadata(gun, NamespacedKey(plugin, "damage"), PersistentDataType.INTEGER, 20)

        // Give player gun
        player.inventory.setItemInMainHand(gun)
    }
}
